--- a/tx.c
+++ b/tx.c
@@ -498,9 +498,11 @@ mt76_txq_send_burst(struct mt76_phy *phy, struct mt76_sw_queue *sq,
 	if (test_bit(MT_WCID_FLAG_PS, &wcid->flags)) {
 		if (!(dev->drv->drv_flags & MT_DRV_HW_PS_BUFFERING))
 			return 0;
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 14, 0)
 		if (ieee80211_txq_aql_pending(phy->hw, txq))
 			return 0;
+#endif
 	}
 
 	if (atomic_read(&wcid->non_aql_packets) >= MT_MAX_NON_AQL_PKT)
